src/stripeActions/process-data/index.js

```diff
@@ -174,99 +77,182 @@ export const processData = async (payload) => {
+              credentials,
+              StripeParams: {
+                id,
+                params: {
+                  metadata: !plans[i][index].planIncludes
+                    ? {}
+                    : plans[i][index].planIncludes,
+                  name: plans[i][index].planType,
+                  description: plans[i][index].description,
+                },
+              },
+            };
+            const updateProductURL = `${
+              SANITY_STUDIO_APP_URL || "https://dxpstudio.webriq.com"
+            }/api/payments/stripe?resource=products&action=update`;
+            try {
+              await fetch(updateProductURL, {
+                headers: {
+                  "Content-Type": "application/json",
+                },
+                method: "POST",
+                body: JSON.stringify(updateProductPayload),
+              });
+            } catch (error) {
+              console.log(error);
+            }
+          } else {
+            const createProductPayload = {
+              credentials,
+              StripeParams: {
+                id,
+                metadata: !plans[i][index].planIncludes
+                  ? {}
+                  : plans[i][index].planIncludes,
+                name: plans[i][index].planType,
+                description: plans[i][index].description,
+              },
+            };
+            const createProductURL = `${
+              SANITY_STUDIO_APP_URL || "https://dxpstudio.webriq.com"
+            }/api/payments/stripe?resource=products&action=create`;
+            try {
+              const response = await fetch(createProductURL, {
+                headers: {
+                  "Content-Type": "application/json",
+                },
+                method: "POST",
+                body: JSON.stringify(createProductPayload),
+              });
+              const createProduct = await response.json();
+              if (createProduct.id) {
+                const createPriceURL = `${
+                  SANITY_STUDIO_APP_URL || "https://dxpstudio.webriq.com"
+                }/api/payments/stripe?resource=prices&action=create`;
 
-                // If Success it will return status = 200 and the data which we will be using is ID to create the price
-                if (responseCreateProduct.meta.status !== 200) {
-                  return {
-                    status: 500,
-                    statusText: `Product Not Created`,
+                if (plans[i].variant === "variant_b") {
+                  const createPricePayload_VariantB = {
+                    credentials,
+                    StripeParams: {
+                      product: createProduct.id,
+                      currency: "usd",
+                      metadata: !plans[i][index].planIncludes
+                        ? {}
+                        : plans[i][index].planIncludes,
+                      unit_amount: plans[i][index].price * 100,
+                    },
                   };
```

Comment:
This payload needs to be updated as per https://github.com/webriq-pagebuilder/app/pull/63#discussion_r719403192